### PR TITLE
DTFS2-4412

### PR DESCRIPTION
**Pull request by eddieleeper**
Add views for UKEF ids and start to secure the application details page.
Only show the details and actions that the user can do depending on their role.

### CODE REVIEW

File: gef-ui/server/controllers/return-to-maker/index.js [hunk 1]
```diff
@@ -20,12 +20,13 @@ export const getReturnToMaker = async (req, res) => {
 };
 
 export const postReturnToMaker = async (req, res, next) => {
-  const { params, body } = req;
-  const { userToken } = req.session;
+  const { params, body, session } = req;
+  console.log(session);
```
Comment (davehlewis) [inline gef-ui/server/controllers/return-to-maker/index.js:24]: bit of debugging kept in?

File: gef-ui/server/controllers/return-to-maker/index.js [hunk 2]
```diff
@@ -39,18 +40,18 @@ export const postReturnToMaker = async (req, res, next) => {
...
         userName: checker.username,
         createdAt: Date.now(),
         comment,
-      },
-    ];
-
-    application.comments = comments;
-
+      };
+      const comments = application.comments || [];
```
Comment (davehlewis) [inline gef-ui/server/controllers/return-to-maker/index.js:50]: was there something wrong with the original implementation here? Needs to be the if there but the rest appears to do the same thing just written in a harder to follow way

File: gef-ui/templates/index.njk [hunk 3]
```diff
@@ -40,9 +40,9 @@
     {% endif %}
 
     {% include "includes/header.njk" %}
+    <div class="govuk-width-container">
```
Comment (davehlewis) [inline gef-ui/templates/index.njk:43]: Does this not cut off the navigation bar? It uses it's own comtainer width I believe?

File: gef-ui/server/controllers/return-to-maker/index.js (hunk 1 above)
Comment (eddieleeper) [reply gef-ui/server/controllers/return-to-maker/index.js:24]: Good spot - I'll remove.

File: gef-ui/server/controllers/return-to-maker/index.js (hunk 2 above)
Comment (eddieleeper) [reply gef-ui/server/controllers/return-to-maker/index.js:50]: The previous implementation always added a comment - even if a comment wasn't typed in the form.  
I'll have a look and see if I can simplify.

File: gef-ui/templates/index.njk (hunk 3 above)
Comment (eddieleeper) [reply gef-ui/templates/index.njk:43]: When I tried it - visually it spanned the whole width of the screen - this seemed to fix it.

**Review by davehlewis — CHANGES_REQUESTED** [review]

File: portal/templates/dashboard/deals.njk [hunk 4]
```diff
@@ -71,8 +71,9 @@
             attributes: { "data-cy": "deal__status--" + deal._id }
           },
           {
-            text: deal.lastUpdate | localiseTimestamp("D MMM YYYY", user.timezone),
-            attributes: { 'data-cy': 'deal__updated--' + deal._id }
+            text: deal.lastUpdate | localiseTimestamp("D MMMYYYY", user.timezone),
```
[the remark this answers is not in the document]
Comment (eddieleeper) [reply portal/templates/dashboard/deals.njk:74]: Excellent - thanks for spotting that @davehlewis

File: gef-ui/server/controllers/application-details/index.js [hunk 5]
```diff
@@ -1,57 +1,119 @@
...
+
+import Application from '../../models/application';
+
+function buildHeader(app) {
+  if (!['Draft'].includes(app.status)) {
+    const main = {
+      ukefDealId: app.ukefDealId || '-',
+      submissionDate: app.submissionDate || '-',
+      companyName: app.exporter.details.companyName,
+      applicationShowSummary: ['SUBMITTED_TO_UKEF', 'CHANGES_REQUIRED', 'BANK_CHECK'].includes(app.status),
```
Comment (ttbarnes) [inline gef-ui/server/controllers/application-details/index.js:18]: should we put `Draft` and `SUBMITTED_TO_UKEF` etc into constants?

File: gef-ui/server/controllers/application-details/index.js [hunk 6]
```diff
@@ -1,57 +1,119 @@
...
+      applicationStatus: app.status,
+      dateCreated: app.createdAt,
+      timezone: app.maker.timezone || 'UTC',
```
Comment (ttbarnes) [inline gef-ui/server/controllers/application-details/index.js:21]: We've historically been setting the timezone to 'Europe/London', is UTC a better way forward? Just thinking of potential confusion for future devs.

File: gef-ui/server/models/application.js [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+      const application = await getApplication(id);
+      if (application.bankId !== user.bank.id) {
+        return null;
+      }
+
+      application.id = id;
+
+      const exporterPro = getExporter(application.exporterId);
+      const coverTermsPro = getCoverTerms(application.coverTermsId);
+      const facilitiesPro = getFacilities(id);
```
[the remark this answers is not in the document]
Comment (eddieleeper) [reply gef-ui/server/models/application.js:23]: Promise - doing it this way so they all run in parallel.

File: gef-ui/templates/includes/application-preview/deal-details.njk [hunk 8]
```diff
@@ -0,0 +1,17 @@
+  <div class="govuk-width-container">
+
+    <div class="govuk-grid-row">
+      <hr width="94%">
+    </div>
+    <div class="govuk-grid-row">
+      <div class="govuk-grid-column-one-half">
+        <div class="govuk-grid-column-one-third"><p><strong>Sent to UKEF</strong></p></div>
+        <div class="govuk-grid-column-two-thirds" data-cy="date-submitted"><p>{{ submissionDate | localiseTimestamp('DD MMM YYYY', timezone) }}</p>
```
Comment (ttbarnes) [inline gef-ui/templates/includes/application-preview/deal-details.njk:9]: Do we need a `dashIfEmpty` filter here or will it always render?

File: gef-ui/server/controllers/application-details/index.js (hunk 6 above)
Comment (eddieleeper) [reply gef-ui/server/controllers/application-details/index.js:21]: Good spot - I'll update.

File: gef-ui/server/models/application.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (ttbarnes) [reply gef-ui/server/models/application.js:23]: Sorry, I didn't see the `Promise.all` afterwards, makes sense

**Review by ttbarnes — APPROVED** [review]
LGTM 🎸

**Review by davehlewis — APPROVED** [review]

**Pull request merged**